openedx/features/personalized_learner_schedules/call_to_action.py

```diff
@@ -96,13 +100,43 @@ def _log_past_due_warning(name):
     @staticmethod
     def _make_reset_deadlines_cta(xblock):
         from lms.urls import RESET_COURSE_DEADLINES_NAME
-        return {
+        request = get_current_request()
+        is_learning_mfe = request and is_request_from_learning_mfe(request)
+        course_key = xblock.scope_ids.usage_id.context_key
+
+        cta_data = {
             'link': reverse(RESET_COURSE_DEADLINES_NAME),
             'link_name': _('Shift due dates'),
             'form_values': {
-                'course_id': xblock.scope_ids.usage_id.context_key,
+                'course_id': course_key,
             },
             'description': _('To participate in this assignment, the suggested schedule for your course needs '
                              'updating. Don’t worry, we’ll shift all the due dates for you and you won’t lose '
                              'any of your progress.'),
+
         }
+
+        if is_learning_mfe:
+            if course_home_mfe_dates_tab_is_active(course_key):
+                body_link = get_microfrontend_url(course_key=course_key, view_name='dates')
+            else:
+                body_link = '{}{}'.format(settings.LMS_ROOT_URL, reverse('dates', args=[six.text_type(course_key)]))
+
+            cta_data['event_data'] = {
+                'body': {
+                    'text': _('View all dates'),
+                    'link': body_link,
```

Comment:
Requiring any body-link to be a full wrapper around the whole body feels limiting. (vs allowing HTML for the body)

I'm guessing you're trying to avoid a dangerouslyInnerHtml in the MFE. But this feels like an API that will require weird carve-outs as soon as we want anything a little different. (Like, if Design wants the next toast to be `<div>Hello <a href="">Nick</a></div>`, do we add an `html` attribute here that *can* be full html and the MFE needs to check for both `text` and `html`?)

If we *do* drop the `link` attribute, we could also theoretically drop the object here and just have `body` be a string field. But you are probably right and we should leave ourselves open for further expansion in the future.